src/jdk.compiler/share/classes/com/sun/tools/javac/jvm/PoolConstant.java

```diff
@@ -228,20 +228,26 @@ public Object poolKey(Types types) {
      */
     final class Parameter implements PoolConstant {
 
+        String id;
         final int kind;
 
         Parameter(int kind) {
```

Comment:
Why do you create a Parameter without an id ?

Reply:
That is during class reading - where we don't care about the "id" anymore. The "id" is just a "fiction" when writing, to help us generating multiple CP holes (possibly with the same kind).

Reply:
Ok, i did not think about that part of the process, and you do not do any roundtrip, reader - >  javac -> gen,
thanks

Reply:
right roundtrips shouldn't happen, this is why we have this duality on the constructors